MSAL/src/native_auth/network/responses/validator/reset_password/MSALNativeAuthResetPasswordResponseValidator.swift

```diff
@@ -126,22 +128,27 @@ final class MSALNativeAuthResetPasswordResponseValidator: MSALNativeAuthResetPas
                 )
             } else {
                 MSALLogger.log(level: .error, context: context, format: "Missing expected fields from backend")
-                return .unexpectedError
+                return .unexpectedError(message: nil)
             }
         case .password,
              .otp:
             MSALLogger.log(level: .error, context: context, format: "ChallengeType not expected")
-            return .unexpectedError
+            return .unexpectedError(message: nil)
         }
     }
 
     private func handleChallengeError(_ error: Error, with context: MSIDRequestContext) -> MSALNativeAuthResetPasswordChallengeValidatedResponse {
         guard let apiError = error as? MSALNativeAuthResetPasswordChallengeResponseError else {
-            MSALLogger.log(level: .info, context: context, format: "Error type not expected")
-            return .unexpectedError
+            MSALLogger.log(level: .info, context: context, format: "Reset Password Challenge: Error was not decoded properly, error: \(error)")
+            return .unexpectedError(message: nil)
+        }
+        switch apiError.error {
```

Comment:
I think using a switch for a single case like this doesn't look right. In this case I think it would be better to keep it simple like

```swift
        if apiError.error == .none {
            return .unexpectedError(message: apiError.errorDescription)
        }

        return .error(apiError)

```
What do you think?

Reply:
Done for all cases when it was the only branch